lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -25,10 +27,15 @@ sub jobs_for_setting ($self, $options) {
 
     # Instead of limiting the number of jobs, this query could also use a trigram gin index to achieve even better
     # performance (at the cost of disk space and setup complexity)
-    $sth = $dbh->prepare(
+    if ($filter_results && $groupids) {
+	$sth = $dbh->prepare(
+        'SELECT job_id, value FROM job_settings INNER JOIN jobs ON jobs.id=job_settings.job_id WHERE key LIKE ? AND value LIKE ? AND jobs.result = ? AND jobs.group_id = ? AND job_id > ? ORDER BY job_settings.job_id DESC');
+	$sth->execute($key_like, $list_value_like, $filter_results, $groupids, $min_id);
+    } else {
+	$sth = $dbh->prepare(
         'SELECT job_id, value FROM job_settings WHERE key LIKE ? AND value LIKE ? AND job_id > ? ORDER BY job_id DESC');
-    $sth->execute($key_like, $list_value_like, $min_id);
-
+	$sth->execute($key_like, $list_value_like, $min_id);
+    }
```

Comment:
The jobgroup id is irrelevant at this stage, if a job matches the search, but it is pre-filtered from the list of results, in the end will hide the data from the user.

Also be aware of the discussion on slack about this route [not returning all of the information](https://suse.slack.com/archives/C02CANHLANP/p1708342424627099?thread_ts=1708020678.269079&cid=C02CANHLANP)